File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.py
class Solution(object):
    def reversePrefix(self, word, ch):
        """
        :type word: str
        :type ch: str
        :rtype: str
        """
        first=""
        second=""
        found=False
        for i in word:
            if found:
                second+=i
            else:
                first+=i
            if i==ch:
                found=True
        if not found:
            return word
        return first[::-1]+second

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        string first="";
        string second="";
        bool found=false;
        for(char i : word){
            if(found){
                second+=i;
            }
            else{
                first+=i;
            }
            if(i==ch){
                found=true;
            }
        }

        if(!found){
            return word;
        }

        int length=first.size()-1;
        string reversed="";
        
        for(int i=length;i>=0;i--){
            reversed+=first[i];
        }

        return reversed+second;
    }
};